src/main/ruby/truffleruby/core/truffle/feature_loader.rb

```diff
@@ -198,11 +198,20 @@ def self.feature_provided?(feature, expanded)
     # MRI: loaded_feature_path
     # Search if $LOAD_PATH[i]/feature corresponds to loaded_feature.
     # Returns the $LOAD_PATH entry containing feature.
-    def self.loaded_feature_path(loaded_feature, feature, load_path)
+    def self.feature_path_loaded?(loaded_feature, feature, load_path)
       name_ext = extension(loaded_feature)
-      load_path.find do |p|
-        loaded_feature == "#{p}/#{feature}#{name_ext}" || loaded_feature == "#{p}/#{feature}"
+
+      suffix_with_ext = "/#{feature}#{name_ext}"
+
+      if name_ext && loaded_feature.end_with?(suffix_with_ext)
```

Comment:
```suggestion
      if name_ext and suffix_with_ext = "/#{feature}#{name_ext}" and loaded_feature.end_with?(suffix_with_ext)
```
Otherwise we compute `suffix_with_ext` even when we don't need it.
And I think it's the most common case to have `name_ext` to be `nil`, because most `require` don't specify extension. But maybe `name_ext` is not as directly related to that, might be good to check.

Reply:
The way I read it is extracting `name_ext` is mainly a safeguard in case `feature` is defined with an extension. Not sure why it's not checked there. I think it makes sense to inline that variable to the condition in any case.